File: Main/ReferenceDLLs/Pack1/Facebook Connect Visual Kit CS/FVK Source Code/FVK/FVK/ComponentParts/ImageData.cs
// WARNING: 
// The source code of this class can be used and modified to your particular needs. 
// Sharing the class with unregistered users or using it with unregistered websites 
// or Facebook applications is not allowed and violates the rights of intellectual 
// property. 

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace FVK
{
    /// <summary>
    /// ImageData class uses to store data about image media for stream publishing
    /// </summary>
    public class ImageData: IJSONSerialize
    {
        /// <summary>
        /// Create new undefined ImageData object
        /// </summary>
        public ImageData() { }

        /// <summary>
        /// Create and define new ImageData object
        /// </summary>
        /// <param name="imageUrl">URL of image</param>
        /// <param name="destUrl">URL of destionation when image is clicked</param>
        public ImageData(string imageUrl, string destUrl)
        {
            this.imageUrl = imageUrl;
            this.destUrl = destUrl;
        }

        /// <summary>
        /// URL of image
        /// </summary>
        public string ImageUrl { get { return imageUrl; } set { imageUrl = value; } }

        /// <summary>
        /// URL of destionation when image is clicked
        /// </summary>
        public string DestUrl { get { return destUrl; } set { destUrl = value; } }

        /// <summary>
        /// Serialize Image data in JSON format
        /// </summary>
        /// <returns>JSON formated string</returns>
        public string JSONSerialize()
        {
            return "{\"type\":\"image\",\"src\":\"" + imageUrl + "\",\"href\":\"" + DestUrl + "\"}";
        }


        /// <summary>
        /// Private members
        /// </summary>
        private string imageUrl;
        private string destUrl;
    }
}

File: Main/ReferenceDLLs/Pack1/Facebook Connect Visual Kit VB/FVK Source Code/FVK/FVK/ComponentParts/ImageData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace FVK
    ''' <summary>
    ''' ImageData class uses to store data about image media for stream publishing
    ''' </summary>
    Public Class ImageData
        Implements IJSONSerialize
        ''' <summary>
        ''' Create new undefined ImageData object
        ''' </summary>
        Public Sub New()
        End Sub

        ''' <summary>
        ''' Create and define new ImageData object
        ''' </summary>
        ''' <param name="imageUrl">URL of image</param>
        ''' <param name="destUrl">URL of destionation when image is clicked</param>
        Public Sub New(ByVal imageUrl As String, ByVal destUrl As String)
            Me.m_imageUrl = imageUrl
            Me.m_destUrl = destUrl
        End Sub

        ''' <summary>
        ''' URL of image
        ''' </summary>
        Public Property ImageUrl() As String
            Get
                Return m_imageUrl
            End Get
            Set(ByVal value As String)
                m_imageUrl = value
            End Set
        End Property

        ''' <summary>
        ''' URL of destionation when image is clicked
        ''' </summary>
        Public Property DestUrl() As String
            Get
                Return m_destUrl
            End Get
            Set(ByVal value As String)
                m_destUrl = value
            End Set
        End Property

        ''' <summary>
        ''' Serialize Image data in JSON format
        ''' </summary>
        ''' <returns>JSON formated string</returns>
        Public Function JSONSerialize() As String _
            Implements IJSONSerialize.JSONSerialize
            Return ("{""type"":""image"",""src"":""" & m_imageUrl & """,""href"":""") + DestUrl & """}"
        End Function


        ''' <summary>
        ''' Private members
        ''' </summary>
        Private m_imageUrl As String
        Private m_destUrl As String
    End Class
End Namespace
